ext/integrations/exec_integration.c

```diff
@@ -0,0 +1,365 @@
+#include "exec_integration.h"
+
+#include <stdio.h>
+#include <symbols/symbols.h>
+#include <sys/wait.h>
+
+#include <ext/standard/file.h>
+#include <ext/standard/proc_open.h>
+
+#include "compatibility.h"
+#include "ddtrace.h"
+#include "span.h"
+
+#define NS "DDTrace\\Integrations\\Exec\\"
+
+#if PHP_VERSION_ID <= 80000
+typedef struct php_process_handle php_process_handle;
+#endif
+
+ZEND_TLS bool in_request;
+
+/* popen stream handler close interception */
+
+static int dd_php_stdiop_close_wrapper(php_stream *stream, int close_handle);
+static bool dd_waitpid(ddtrace_span_data *, php_process_id_t);
+
+ZEND_TLS HashTable *tracked_streams;  // php_stream => span
+static zend_string *cmd_exit_code_zstr;
+static zend_string *error_message_zstr;
+static zend_string *has_signalled_zstr;
+static zend_string *pclose_minus_one_zstr;
+static int (*orig_php_stream_stdio_ops_close)(php_stream *stream, int close_handle);
+
+static void dd_exec_lazy_init_track_streams() {
+    if (tracked_streams) {
+        return;
+    }
+
+    ALLOC_HASHTABLE(tracked_streams);
+    zend_hash_init(tracked_streams, 8, NULL, ZVAL_PTR_DTOR, 0);
+}
```

Comment:
What's the reasoning behind lazy hashtable init? The overhead here is negligible, I'd just always init it.
Allows also for simplifying the code of just using `if(tracked_streams)` rather than the extra in_request bool.

Reply:
I followed your suggestion.